src/Workspaces/Remote/Core/RemoteEndPoint.cs

```diff
@@ -191,16 +191,20 @@ public async Task<T> InvokeAsync<T>(string targetName, IReadOnlyList<object?> ar
                 // if invoke throws an exception, make sure we raise cancellation.
                 RaiseCancellationIfInvokeFailed(task, linkedCancellationSource, cancellationToken);
 
-                // wait for asset source to respond
-                await pipe.WaitForConnectionAsync(linkedCancellationSource.Token).ConfigureAwait(false);
+                var task2 = Task.Run(async () =>
+                {
+                    // wait for asset source to respond
+                    await pipe.WaitForConnectionAsync(linkedCancellationSource.Token).ConfigureAwait(false);
 
-                // run user task with direct stream
-                var result = await dataReader(stream, linkedCancellationSource.Token).ConfigureAwait(false);
+                    // run user task with direct stream
+                    return await dataReader(stream, linkedCancellationSource.Token).ConfigureAwait(false);
+                });
 
-                // wait task to finish
-                await task.ConfigureAwait(false);
+                // wait for tasks to finish
+                await Task.WhenAll(task, task2).ConfigureAwait(false);
 
-                return result;
+                Debug.Assert(task2.IsCompletedSuccessfully);
```

Comment:
No need for this at all. Because `Task.WhenAll` *throws* if *any* of the tasks do not complete successfully.

Reply:
It's intended as a communication to the reader.